clab/config/template.go

```diff
@@ -68,22 +68,25 @@ func RenderAll(allnodes map[string]*NodeConfig) error {
 	tmpl := template.New("").Funcs(jT.Funcs)
 
 	for _, nc := range allnodes {
-
 		for _, baseN := range TemplateNames {
 			tmplN := fmt.Sprintf("%s__%s.tmpl", baseN, nc.Vars[vkRole])
-
-			if tmpl.Lookup(tmplN) == nil {
+			log.Debugf("Looking up template %v", tmplN)
+			if l := tmpl.Lookup(tmplN); l == nil {
 				err := LoadTemplates(tmpl, fmt.Sprintf("%s", nc.Vars[vkRole]))
 				if err != nil {
-					return err
+					log.Warnf("Unable to load template %s; skipping", tmplN)
+					continue
 				}
-				if tmpl.Lookup(tmplN) == nil {
-					return fmt.Errorf("template not found %s", tmplN)
+				l = tmpl.Lookup(tmplN)
```

Comment:
is this assignment needed, considering that `l` was assigned on L74?

If it is needed, then assignment on L74 is not?

Reply:
This one is needed, the logic as follows: 1) check if we have a template in memory for a given node type X (L74); 2) if not - try to load it; (L75) 3) check again (the result) (L80-82). So if 2) gives us non-nil result, we'll be proceeding further, otherwise skipping an iteration based on check 3). We can eliminate the first assignment indeed, and I've considered it, but it will bring some extra work for nodes with templates (available templates will be loaded from a disk for each node vs once per kind). I don't have a strong preference here so kept the current logic with more code / fewer computations. Let me know if you want to change this.

Reply:
okay, lets take it one step at a time